From 96b1432e8af4153f8d94a6be62b2538c999c9f96 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Mon, 11 Jan 2021 08:39:05 +0200
Subject: [PATCH] [SW-0] i2c: config access: use regmap

Use regmap_readl/writel for config access - no functionality changes

Change-Id: I878c7f820a5fdf15d33e5fc322b9f00cd20358ad
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/preboot/i2cslave_indirect.c b/preboot/i2cslave_indirect.c
index a0f5ae0..48f1d1a 100644
--- a/preboot/i2cslave_indirect.c
+++ b/preboot/i2cslave_indirect.c
@@ -62,10 +62,10 @@
 {
 	switch (op_config.command) {
 	case HL_I2CI_CONFIG_CMD_READ:
-		op_config.data = *(u32 *)(HL_CFG_BASE + op_config.address);
+		op_config.data = regmap_readl(op_config.address);
 		break;
 	case HL_I2CI_CONFIG_CMD_WRITE:
-		*(u32 *)(HL_CFG_BASE + op_config.address) = op_config.data;
+		regmap_writel(op_config.data, op_config.address);
 		break;
 	}
 
